src/Sarif.Viewer.VisualStudio/ProjectNameLookup.cs

```diff
@@ -0,0 +1,33 @@
+﻿using System.Collections.Generic;
+
+namespace Microsoft.Sarif.Viewer
+{
+    public sealed class ProjectNameLookup
+    {
+        private readonly static ProjectNameLookup _instance = new ProjectNameLookup();
+        private Dictionary<string, string> _projectNames = new Dictionary<string, string>();
+
+        static ProjectNameLookup() { }
+        private ProjectNameLookup() { }
```

Comment:
I was referring to the `static` constructor.

The `private` constructor is indeed part of the Singleton pattern. It's not there to allow the Singleton to be instantiated; it's there to _prevent_ anyone from creating additional instances by writing `new Foo()`.

---
In reply to: [135286787](https://github.com/Microsoft/sarif-sdk/pull/596#discussion_r135286787) [](ancestors = 135286787)

Reply:
I was going for the fourth implementation found here: [http://csharpindepth.com/Articles/General/Singleton.aspx](http://csharpindepth.com/Articles/General/Singleton.aspx).  Classes with a static constructor apparently get their type initializers invoked only once per app domain which prevents us from having to use a null check or lock to ensure thread safety.  More information on this can be found here:  [http://csharpindepth.com/Articles/General/BeforeFieldInit.aspx](http://csharpindepth.com/Articles/General/BeforeFieldInit.aspx).  Do you think we should change this to use a lock instead?

---
In reply to: [135287508](https://github.com/Microsoft/sarif-sdk/pull/596#discussion_r135287508) [](ancestors = 135287508,135286787)

Reply:
I did not know about this! No, keep what you have, it is fine.

---
In reply to: [135301008](https://github.com/Microsoft/sarif-sdk/pull/596#discussion_r135301008) [](ancestors = 135301008,135287508,135286787)